pynemo/nemo_bdy_extr_tm3.py

```diff
@@ -600,32 +654,48 @@ def extract_month(self, year, month):
 
             # Calculate weightings to be used in interpolation from
             # source data to dest bdy pts. Only need do once.
-            if self.first:
+            #if self.first:
+            for vn in range(self.nvar):
+
+
+                varid = sc_time[self.var_nam[vn]]
+                # Determine if slab or not
+                isslab = len(varid._get_dimensions())==3
+
+                # Set up tmp dict of tmp arrays
+                if isslab:
+                    sc_z_len=1
+                else:
+                    sc_z_len=self.sc_z_len
+
```

Comment:
differentiate between 2D and 3D .... this naively assumes there is always a time dimension... something that is mostly true, but will have to be tidied up